feature/tabsflow/moreflow/faq/src/main/java/com/epmedu/animeal/tabs/more/faq/presentation/viewmodel/FAQViewModel.kt

```diff
@@ -1,21 +1,25 @@
-package com.epmedu.animeal.tabs.more.faq.viewmodel
+package com.epmedu.animeal.tabs.more.faq.presentation.viewmodel
 
 import androidx.lifecycle.ViewModel
-import com.epmedu.animeal.common.component.BuildConfigProvider
+import androidx.lifecycle.viewModelScope
 import com.epmedu.animeal.common.presentation.viewmodel.delegate.DefaultStateDelegate
 import com.epmedu.animeal.common.presentation.viewmodel.delegate.StateDelegate
+import com.epmedu.animeal.tabs.more.faq.domain.GetFAQUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 internal class FAQViewModel @Inject constructor(
-    private val buildConfigProvider: BuildConfigProvider
+    private val getFAQUseCase: GetFAQUseCase
 ) : ViewModel(),
     StateDelegate<FAQState> by DefaultStateDelegate(initialState = FAQState()) {
 
     init {
-        updateState {
-            copy(appVersionName = buildConfigProvider.appVersion)
+        viewModelScope.launch {
```

Comment:
What do you think about extracting it to a function for future readability?

Reply:
Sure